tests/components/template/test_fan.py

```diff
@@ -222,6 +222,38 @@ async def test_templates_with_entities(hass, calls):
     _verify(hass, STATE_ON, SPEED_MEDIUM, True, DIRECTION_FORWARD)
 
 
+async def test_template_with_unavailable_entities(hass, calls):
+    """Test availability tempalates with values from other entities."""
+
+    with assert_setup_component(1, "fan"):
+        assert await setup.async_setup_component(
+            hass,
+            "fan",
+            {
+                "fan": {
+                    "platform": "template",
+                    "fans": {
+                        "test_fan": {
+                            "availability_template": "{{ is_state('availability_boolean.state', 'on') }}",
+                            "value_template": "{{ 'unavailable' }}",
+                            "speed_template": "{{ 'unavailable' }}",
+                            "oscillating_template": "{{ 'unavailable' }}",
+                            "direction_template": "{{ 'unavailable' }}",
+                            "turn_on": {"service": "script.fan_on"},
+                            "turn_off": {"service": "script.fan_off"},
+                        }
+                    },
+                }
+            },
+        )
+
+    await hass.async_block_till_done()
+    await hass.async_start()
+    await hass.async_block_till_done()
+
+    _verify(hass, STATE_UNAVAILABLE, None, None, None)
```

Comment:
We need at least two separate tests, one for state template which renders to unavailable, and a different test for speed, oscillating, and direction since when the state is unavailable, the other attributes aren't updated for the entity.